File: AlgoSolution/uva/uva465.cpp
#include <cstdio>
#include <cstring>
#include <iostream>
#include <climits>

using namespace std;

int main(){
    char add1[100], add2[100], op[10];
    while(scanf("%s %s %s", add1, op, add2) != EOF){
     int a, b;
     a = b = 0;
     printf("%s %s %s\n", add1, op, add2);
     if(add1[0] == '-') printf("%c\n", op[-1]);
     // 貌似没有负数
     for(int i = 0; i < strlen(add1); i ++){
        a = a*10 + (add1[i]-'0');
     }
     for(int i = 0; i < strlen(add2); i ++){
        b = b*10 + (add2[i]-'0');
     }
     //cout << (long long)a*b << endl;
     int flag1, flag2;
     flag1 = flag2 = 1;
     if(a > INT_MAX || a < 0) {
            printf("first number too big\n");
            flag1 = 0;
     }
     if(b > INT_MAX || b < 0) {
            printf("second number too big\n");
            flag2 = 0;
     }
     if(strcmp(op, "+") && ( a+b < 0 ||  flag1 == 0 || flag2 == 0))
        printf("result too big\n");
     if(strcmp(op, "*") && ( a*b < 0 ||  flag1 == 0 || flag2 == 0))
        printf("result too big\n");
    }
    return 0;
}

File: AlgoSolution/uva/uva465.java
import java.math.*;
import java.util.*;

public class Test {
	public static void main(String args[]){
		Scanner cin = new Scanner(System.in);
		BigInteger x, y, Max;
		Max = new BigInteger(Integer.toString(Integer.MAX_VALUE));
		String s, a, b, c;
		while(cin.hasNextLine()){
			s = cin.nextLine();
			System.out.println(s);
			int i = 0;
			a = b = c = "";
			for(i = 0; ; i ++){
				if(s.charAt(i) == '*' || s.charAt(i) == '+') break;
				if(s.charAt(i) == ' ') continue;
				a += s.charAt(i);
			}
			b += s.charAt(i);
			i ++;
			for(; i < s.length(); i ++){
				if(s.charAt(i) == ' ') continue;
				c += s.charAt(i);
			}
			x = new BigInteger(a);
			y = new BigInteger(c);
			if(x.compareTo(Max) > 0) System.out.println("first number too big");
			if(y.compareTo(Max) > 0) System.out.println("second number too big");
			if(b.equals("+") && x.add(y).compareTo(Max) > 0 )
				System.out.println("result too big");
			if(b.equals("*") && x.multiply(y).compareTo(Max) > 0)
				System.out.println("result too big");
		}
		cin.close();
	}
}

/*
conclusion:
13777396 	465 	Overflow
Accepted 	JAVA 	0.178 	2014-06-22 02:29:42
1: 选择etLine多行输入比较好
2: bigInteger 构造函数中字符不能用
*/
